### PR TITLE
[Swift] Silence false positives in ASAN.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/50644029>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

**Comment by dcci** [comment]
@adrian-prantl can you please take a look at this before it goes in? :)

**Review by adrian-prantl — APPROVED** [review]

File: lit/lit.cfg.py [hunk 1]
```diff
@@ -35,6 +35,12 @@
 # test_exec_root: The root path where tests should be run.
 config.test_exec_root = os.path.join(config.lldb_obj_root, 'lit')
 
+# Begin Swift mod.
+# Swift's libReflection builds without ASAN, which causes a known
+# false positive in std::vector. If sanitizers are off, this is just
+# a no-op
+config.environment['ASAN_OPTIONS'] = ':detect_container_overflow=0'
```
Comment (JDevlieghere) [inline lit/lit.cfg.py]: Spurious `:`?

Comment (dcci) [reply lit/lit.cfg.py]: Done.

**Comment by dcci** [comment]
@swift-ci test and merge

**Pull request merged**